packages/local_auth/local_auth_platform_interface/lib/default_method_channel_platform.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:local_auth_platform_interface/local_auth_platform_interface.dart';
+import 'package:local_auth_platform_interface/types/auth_options.dart';
+import 'package:local_auth_platform_interface/types/biometric_type.dart';
+
+const MethodChannel _channel = MethodChannel('plugins.flutter.io/local_auth');
+
+/// The default interface implementation acting as a placeholder for
+/// the native implementation to be set.
+class DefaultLocalAuthPlatform extends LocalAuthPlatform {
+  @override
+  Future<bool> authenticate({
+    required String localizedReason,
+    required Map<String, String> authStrings,
+    AuthenticationOptions options = const AuthenticationOptions(),
+  }) async {
+    assert(localizedReason.isNotEmpty);
+    final Map<String, Object> args = <String, Object>{
+      'localizedReason': localizedReason,
+      'useErrorDialogs': options.useErrorDialogs,
+      'stickyAuth': options.stickyAuth,
+      'sensitiveTransaction': options.sensitiveTransaction,
+      'biometricOnly': options.biometricOnly,
+    };
+    args.addAll(authStrings);
+    return (await _channel.invokeMethod<bool>('authenticate', args)) ?? false;
+  }
+
+  @override
+  Future<List<BiometricType>> getAvailableBiometrics() async {
+    final List<String> result = (await _channel.invokeListMethod<String>(
+          'getAvailableBiometrics',
+        )) ??
+        <String>[];
+    final List<BiometricType> biometrics = <BiometricType>[];
+    for (final String value in result) {
+      switch (value) {
+        case 'face':
+          biometrics.add(BiometricType.face);
+          break;
+        case 'fingerprint':
+          biometrics.add(BiometricType.fingerprint);
+          break;
+        case 'iris':
+          biometrics.add(BiometricType.iris);
+          break;
+        case 'undefined':
+          break;
+      }
+    }
+    return biometrics;
+  }
+
```

Comment:
Don't you need `canCheckBiometrics` in the platform interface, given this implementation? The way `canCheckBiometrics` works right now (on iOS; it's [currently broken on Android](https://github.com/flutter/plugins/pull/4682)) is that `'undefined'` signals that the device has biometrics capability, but nothing is enrolled. Since you're filtering at this layer, that can't be implemented in the app-facing plugin.

(I think that implementation is super-confusing, so I'm all for having `canCheckBiometrics` being in the platform interface so that implementations can move over time to something less weird without any changes to the interface.)

Reply:
Ah, I see. That behaviour is indeed a bit odd. I'll add it to the platform interface to allow for a custom implementation. 

For now I'll keep the same implementation in the iOS and Android packages, but I'll look into submitting a separate PR with a fix after they have been merged.